DependencyInjection/HttplugExtension.php

```diff
@@ -430,33 +417,46 @@ private function configureAutoDiscoveryClients(ContainerBuilder $container, arra
     /**
      * Find a client with auto discovery and return a service Reference to it.
      *
-     * @param ContainerBuilder $container
-     * @param string           $name
-     * @param callable         $factory
-     * @param bool             $profiling
+     * @param ContainerBuilder   $container
+     * @param string             $name
+     * @param Reference|callable $factory
+     * @param bool               $profiling
      *
      * @return string service id
      */
     private function registerAutoDiscoverableClient(ContainerBuilder $container, $name, $factory, $profiling)
     {
         $serviceId = 'httplug.auto_discovery.'.$name;
 
-        $pluginClientOptions = [];
-
+        $plugins = [];
         if ($profiling) {
-            // Tell the plugin journal what plugins we used
-            $container
-                ->getDefinition('httplug.collector.plugin_journal')
-                ->addMethodCall('setPlugins', [$name, ['httplug.plugin.stopwatch']])
-            ;
+            // Add the newstack plugin
```

Comment:
new stack plugin? or newstack? if i guess correctly what we do here, lets do: `// To profile requests, we put a stack plugin instance as first plugin in the chain, followed by the stopwatch plugin`